kotlintest-core/src/commonMain/kotlin/io/kotlintest/core/specs/AbstractSpecDsl.kt

```diff
@@ -0,0 +1,106 @@
+package io.kotlintest.core.specs
+
+import io.kotlintest.AssertionMode
+import io.kotlintest.IsolationMode
+import io.kotlintest.Spec
+import io.kotlintest.Tag
+import io.kotlintest.TestCase
+import io.kotlintest.TestCaseOrder
+import io.kotlintest.TestResult
+import io.kotlintest.extensions.SpecLevelExtension
+import io.kotlintest.extensions.TestListener
+
+abstract class AbstractSpecDsl : AbstractSpec() {
+
+   private var beforeTestFn: (TestCase) -> Unit = {}
+   private var afterTestFn: (TestCase, TestResult) -> Unit = { _, _ -> }
+   private var beforeSpecFn: (Spec) -> Unit = {}
+   private var afterSpecFn: (Spec) -> Unit = {}
+   private var afterSpecClassFn: (Spec, Map<TestCase, TestResult>) -> Unit = { _, _ -> }
+   private var tags: Set<Tag> = emptySet()
+   private var listeners: List<TestListener> = emptyList()
+   private var extensions: List<SpecLevelExtension> = emptyList()
+   private var testCaseOrder: TestCaseOrder? = null
+   private var isolationMode: IsolationMode? = null
+   private var assertionMode: AssertionMode? = null
+
+   override fun testCaseOrder(): TestCaseOrder? = testCaseOrder
+   override fun isolationMode(): IsolationMode? = isolationMode
+   override fun assertionMode(): AssertionMode? = assertionMode
+
+   override fun listeners(): List<TestListener> = listeners
+   override fun extensions(): List<SpecLevelExtension> = extensions
+
+   override fun beforeTest(testCase: TestCase) {
+      super.beforeTest(testCase)
+      beforeTestFn(testCase)
```

Comment:
Perhaps this is kind of dangerous. It wouldn't be possible to mix old syntax with new syntax this way (maybe we don't want to?), as any override could not call `super`

Reply:
An override could call super, but no one is going to override before test in both places.